test/tests/functional/pbs_hook_execjob_prologue.py

```diff
@@ -236,3 +236,22 @@ def test_prologue_hook_fail_action_non_mom_hook(self):
             exp_err += " contain at least one of execjob_begin"
             exp_err += ", exechost_startup, execjob_prologue"
             self.assertTrue(exp_err in e.msg[0])
+
+    def test_prologue_hook_does_not_execute_twice_on_pbsdsh(self):
+        """
+        """
+        hook_name = 'prologue'
+        hook_body = ("import pbs\n"
+                     "e = pbs.event()\n"
+                     "pbs.logjobmsg(e.job.id, 'executed prologue hook')\n")
+        attr = {'event': 'execjob_prologue'}
+        self.server.create_import_hook(hook_name, attr, hook_body)
+
+        j = Job(TEST_USER, {'Resource_List.select': '2:ncpus=1',
+                            'Resource_List.place': 'scatter'})
+        j.create_script('pbsdsh -n 1 hostname')
+        jid = self.server.submit(j)
+
+        match = self.momB.log_match("Job;%s;executed prologue hook" % jid,
+                                    n=100, max_attempts=10, interval=2)
+        self.assertTrue(isinstance(match, tuple))
```

Comment:
This isn't actually useful.  Either the log_match() will fail and raise a PtlLogMatchError or it will return a tuple.  I think what you want to do is check the second element in the tuple for 1.  The tuple is (matched_line, number_of_matches).  You should check number_of_matches to be 1.

Reply:
The tuple `log_match()` returns is (matched_line_number, the line itself). So I cannot check the second part of the tuple.

Reply:
I updated the test to do another assert on `match is not None`. I don't know if that helps.